src/NServiceBus.Core/DelayedDelivery/TimeoutManager/TimeoutManager.cs

```diff
@@ -65,45 +65,50 @@ static void SetupTimeoutPoller(FeatureConfigurationContext context, string dispa
 
         static string SetupDispatcherSatellite(FeatureConfigurationContext context, string errorQueueAddress, TransportTransactionMode requiredTransactionSupport)
         {
-            string dispatcherAddress;
-            var dispatcherProcessorPipeline = context.AddSatellitePipeline("Timeout Dispatcher Processor", requiredTransactionSupport, PushRuntimeSettings.Default, "TimeoutsDispatcher", out dispatcherAddress);
-
-            dispatcherProcessorPipeline.Register("DispatchTimeoutRecoverability",
-                b => CreateTimeoutRecoverabilityBehavior(errorQueueAddress, dispatcherAddress, b),
-                "Handles failures when dispatching timeouts");
-
-            dispatcherProcessorPipeline.Register("TimeoutDispatcherProcessor", b => new DispatchTimeoutBehavior(
-                b.Build<IDispatchMessages>(),
-                b.Build<IPersistTimeouts>(),
-                requiredTransactionSupport),
-                "Terminates the satellite responsible for dispatching expired timeouts to their final destination");
-            return dispatcherAddress;
+            var instanceName = context.Settings.EndpointInstanceName();
+            var satelliteLogicalAddress = new LogicalAddress(instanceName, "TimeoutsDispatcher");
+            var satelliteAddress = context.Settings.GetTransportAddress(satelliteLogicalAddress);
+            var failureStorage = new TimeoutFailureInfoStorage();
+
+            context.AddSatelliteReceiver("Timeout Dispatcher Processor", satelliteAddress, requiredTransactionSupport, PushRuntimeSettings.Default, (builder, pushContext) =>
+            {
+                var recBehavior = CreateTimeoutRecoverabilityBehavior(errorQueueAddress, satelliteAddress, builder, failureStorage);
+                var dispatchBehavior = new DispatchTimeoutBehavior(
+                    builder.Build<IDispatchMessages>(),
+                    builder.Build<IPersistTimeouts>(),
+                    requiredTransactionSupport);
+
+                return recBehavior.Invoke(pushContext, () => dispatchBehavior.Invoke(pushContext));
```

Comment:
Looks like that was done to get access to the builder? Would there be some other way to provide access here so that they wouldn't need to be created for every message?